cincoconfig/config.py

```diff
@@ -309,6 +309,33 @@ def wrapper(meth: ConfigInstanceMethod) -> ConfigInstanceMethod:
             return meth
         return wrapper
 
+    def generate_argparse_parser(self, **kwargs) -> ArgumentParser:
+        '''
+        Generate a :class:`argparse.ArgumentParser` based on the schema. This method generates
+        ``--long-arguments`` for each field that stores a string, integer, float, or bool (based
+        on the field's ``storage_type``). Boolean fields have two long arguments created, one to
+        store a ``True`` value and another, ``--no-[x]``, to disable it.
+
+        :param kwargs: keyword arguments to pass to the generated ``ArgumentParser`` constructor
+        :returns: the generated argument parser
+        '''
+        parser = ArgumentParser(**kwargs)
+        for name, _, field in self.get_all_fields():
+            if not isinstance(field, Field):
+                continue
+
+            arg = '--' + name.replace('.', '-').replace('_', '-').lower()
+            metavar = name.replace('.', '_').upper()
+            if field.storage_type in (str, float, int):
+                parser.add_argument(arg, action='store', dest=name, help=field.short_help,
+                                    metavar=metavar)
```

Comment:
Would you consider adding support for environment variables to argparse? (particularly useful for secrets in pipelines or containerized environments)

```suggestion
            long_arg = '--' + name.replace('.', '-').replace('_', '-').lower()
            env_var_name = name.replace('.', '_').upper()
            metavar = name.replace('.', '_').upper()
            if field.storage_type in (str, float, int):
                parser.add_argument(long_arg, action='store', dest=name, help=field.short_help,
                                    metavar=metavar, default=os.environ.get(env_var_name))
```

Reply:
Hmmm, is there a reason you'd want to set the default in the `ArgumentParser` and not the Field itself?

```python
schema.password = FIeld(default=os.environ.get('DATABASE_PASSWORD'))
```

If an argument isn't specified it'll use the `Field.default`.

Reply:
Yes, the default value in the config schema should be the application default, ex:

```py
Schema.log_level_console = LogLevelField(default='info')
```

You could do `default=os.environ.get(metavar, 'info')` I suppose, but I thought it would be nice to have the environment variable code auto-generated, so the whole config isn't filled with `os.environ.get('CAP_STR')` entries.

Setting it in argparse isn't actually setting a default, since `os.environ.get()` will return NONE if the variable isn't defined, and therefore the config item won't be overridden in `cmdline_args_override`